consensus/src/payload_manager.rs

```diff
@@ -16,17 +17,60 @@ use aptos_logger::prelude::*;
 use aptos_types::transaction::SignedTransaction;
 use futures::channel::mpsc::Sender;
 use std::sync::Arc;
-use tokio::sync::oneshot;
+use tokio::sync::{mpsc::UnboundedSender, oneshot};
 
 pub trait TPayloadManager: Send + Sync {
     fn prefetch_payload_data(&self, payload: &Payload, timestamp: u64);
 }
 
+pub struct DAGLink {
+    dag_payload_link_tx: UnboundedSender<PayloadLinkMsg>,
+}
+
+impl DAGLink {
+    #[allow(unused)]
+    pub fn new(tx: UnboundedSender<PayloadLinkMsg>) -> Self {
+        Self {
+            dag_payload_link_tx: tx,
+        }
+    }
+
+    async fn get_transactions(
+        &self,
+        bundle: DAGPayloadBundle,
+    ) -> ExecutorResult<Vec<SignedTransaction>> {
+        debug!("get_transactions {:?}", bundle);
+        let len = bundle.len();
```

Comment:
this is confusing, I thought it's the number of batches, probably call it num_txns